### PR TITLE
python311Packages.crytic-compile: 0.3.3 -> 0.3.4

### PR DESCRIPTION
**Pull request by hellwolf**
## Description of changes

    python311Packages.crytic-compile: 0.3.3 -> 0.3.4

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hellwolf** [comment]
Result of `nixpkgs-review pr 247944` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>echidna</li>
    <li>python310Packages.crytic-compile</li>
    <li>python310Packages.crytic-compile.dist</li>
    <li>slither-analyzer (python310Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python310Packages.slither-analyzer.dist)</li>
    <li>python311Packages.crytic-compile</li>
    <li>python311Packages.crytic-compile.dist</li>
    <li>python311Packages.slither-analyzer</li>
    <li>python311Packages.slither-analyzer.dist</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 247944` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>echidna</li>
    <li>python310Packages.crytic-compile</li>
    <li>python310Packages.crytic-compile.dist</li>
    <li>slither-analyzer (python310Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python310Packages.slither-analyzer.dist)</li>
    <li>python311Packages.crytic-compile</li>
    <li>python311Packages.crytic-compile.dist</li>
    <li>python311Packages.slither-analyzer</li>
    <li>python311Packages.slither-analyzer.dist</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
There is a hash mismatch in one of the dependency's

**Comment by hellwolf** [comment]
> There is a hash mismatch in one of the dependency's 

Strange, I ran the same thing. See my previous post. I will try again.

Any more details you could share?

**Comment by hellwolf** [comment]
> There is a hash mismatch in one of the dependency's

I have just rebased this PR, and did it again:

Result of `nixpkgs-review pr 247944` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>echidna</li>
    <li>python310Packages.crytic-compile</li>
    <li>python310Packages.crytic-compile.dist</li>
    <li>slither-analyzer (python310Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python310Packages.slither-analyzer.dist)</li>
    <li>python311Packages.crytic-compile</li>
    <li>python311Packages.crytic-compile.dist</li>
    <li>python311Packages.slither-analyzer</li>
    <li>python311Packages.slither-analyzer.dist</li>
  </ul>
</details>


Everything works, so please provide more details on your end. Meanwhile let me ask: 

@NixOS/ofborg  build echidna python310Packages.crytic-compile python311Packages.crytic-compile python311Packages.slither-analyzer

**Comment by hellwolf** [comment]
@NixOS/ofborg  build echidna python310Packages.crytic-compile python311Packages.crytic-compile python311Packages.slither-analyzer

**Comment by hellwolf** [comment]
@ofborg build echidna python310Packages.crytic-compile python311Packages.crytic-compile python311Packages.slither-analyzer

**Review by drupol — APPROVED** [review]

File: pkgs/development/python-modules/crytic-compile/default.nix [hunk 1]
```diff
@@ -19,7 +19,7 @@ buildPythonPackage rec {
     owner = "crytic";
     repo = "crytic-compile";
     rev = "refs/tags/${version}";
-    hash = "sha256-Nx3eKy/0BLg82o3qDHjxcHXtpX3KDdnBKYwCuTLWRUE=";
+    hash = "sha256-b1mAiua+9twXRN6sETednmH24TzZAKOiIIxDSA4mrlc=";
```
Comment (drupol) [inline pkgs/development/python-modules/crytic-compile/default.nix:22]: Shouldn't it be this ?
```suggestion
    hash = "sha256-CeoACtgvMweDbIvYguK2Ca+iTBFONWcE2b0qUkBbQSU=";
```

**Review by drupol — CHANGES_REQUESTED** [review]
Hash mismatch, please update it.

File: pkgs/development/python-modules/crytic-compile/default.nix (hunk 1 above)
Comment (hellwolf) [reply pkgs/development/python-modules/crytic-compile/default.nix]: Thank you! Indeed, I think the upstream changed their artifact. I was probably too fast.

**Comment by hellwolf** [comment]
> Hash mismatch, please update it.

done.

I guess if upstream change their artifact with the same rev reference, the nixpkgs cache would unfortunately not detect this error?

**Pull request merged**